### PR TITLE
Configure Gitpod for one-click quickstart

### PR DESCRIPTION
**Pull request by jacobPARIS**
In this PR I've set up Gitpod with a custom Dockerfile that comes with Fly pre-installed. The goal is for people following the blog tutorial to not have to worry about setting up their local environment or installing Fly. In one click, they should be able to get into a code editor with the demo application running.

The first time a workspace is opened, a `.env` will be created automatically based on the `.env.example`. If the same workspace is closed and re-opened, local changes to this file will be persisted, but it will **not** be shared between workspaces. 

If you do want to set a `.env` file and have it be the default for all new workspaces in this repo (useful if you're working on this repo itself or just often use it to start new projects with a similar config), the following command will serialize changes to the`.env` file to Gitpod's Workspace Environment Variables and future workspaces will load from that instead of the `.env.example`

```sh
eval $(gp env -e ENV="$(base64 .env | tr -d '\n')")
```

You can test this workflow by clicking this button:

[Gitpod Ready-to-Code](https://gitpod.io/#https://github.com/jacobPARIS/indie-stack/)

### CODE REVIEW

File: .gitpod.yml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    - ms-azuretools.vscode-docker
+    - esbenp.prettier-vscode
+    - dbaeumer.vscode-eslint
+    - eamodio.gitlens
+    - bradlc.vscode-tailwindcss
+    - oderwat.indent-rainbow
+    - wix.vscode-import-cost
+    - christian-kohler.npm-intellisense
+    - christian-kohler.path-intellisense
+    - csstools.postcss
```
Comment (kentcdodds) [inline .gitpod.yml:54]: I'd prefer to only include the plugins that are absolutely necessary:

```suggestion
    - ms-azuretools.vscode-docker
    - esbenp.prettier-vscode
    - dbaeumer.vscode-eslint
    - bradlc.vscode-tailwindcss
```

File: README.md [hunk 2]
```diff
@@ -26,8 +26,15 @@ npx create-remix --template remix-run/indie-stack
...
 
+## Quickstart
+
+Click this button to create a [Gitpod](https://gitpod.io) workspace with the project set up and Fly pre-installed
+
+[![Gitpod Ready-to-Code](https://img.shields.io/badge/Gitpod-Ready--to--Code-blue?logo=gitpod)](https://gitpod.io/from-referrer/)
+
 ## Development
 
+
```
Comment (kentcdodds) [inline README.md:37]:
```suggestion
```

**Review by kentcdodds — CHANGES_REQUESTED** [review]
Looks good. Just a couple things.

**Comment by jacobPARIS** [comment]
> I'd prefer to only include the plugins that are absolutely necessary

Agreed, and I've adjusted the list to your recommendation. I've got a good batch of extensions I like working with, and I figured it was easier to remove the ones we don't want than to come up with the ones we do

From @andrewbrey's comment https://github.com/remix-run/remix/pull/2711
> @jacobPARIS I just gave this a try against your repository branch, and noted the prisma dev migration is not done automatically at container init. That might be a nice thing to add to the setup script (like you're doing when you create the `.env` file for the user). If you did that, clicking "open in gitpod" would launch into a fully working and initialized dev environment, db and all 👍

This was a great suggestion, and that's been implemented now too. A new user can spin up a workspace and immediately create an account with no Prisma errors

**Review by kentcdodds — APPROVED** [review]
Coolio 👍 Thanks for putting this together!

**Comment by MichaelDeBoey** [comment]
@jacobPARIS Would be lovely to have these in [`blues-stack`](https://github.com/remix-run/blues-stack) & [`grunge-stack`](https://github.com/remix-run/grunge-stack) as well

**Comment by jacobPARIS** [comment]
> @jacobPARIS Would be lovely to have these in [`blues-stack`](https://github.com/remix-run/blues-stack) & [`grunge-stack`](https://github.com/remix-run/grunge-stack) as well

I will make that happen :)

**Pull request merged**